conventional_precommit_linter/hook.py

```diff
@@ -0,0 +1,182 @@
+
+def parse_commit_message(args: argparse.Namespace, input_commit_message: str) -> None:
+    # Split the 'commit message' into the 'message title' (first line) and 'message body' (the rest)
+    message_title, *message_body = input_commit_message.strip().split('\n', 1)
+
+    # First split 'message title' into potential 'type/scope' and 'summary'
+    message_parts = message_title.split(': ', 1)  # using 1 as second argument to split only on first occurrence
+    if len(message_parts) != 2:
+        types = regex_types(args)
+        raise_error(message_title, ERROR_MISSING_COLON, types, args)
+
+    # Check if a 'scope' is provided in the potential 'type/scope' part
+    type_scope_parts = message_parts[0].split('(', 1)  # using 1 as second argument to split only on first occurrence
+    if len(type_scope_parts) == 1:
+        # no 'scope' provided
+        commit_type = type_scope_parts[0]
+        commit_scope = None
+    else:
+        # 'scope' provided
+        commit_type = type_scope_parts[0]
+        commit_scope = type_scope_parts[1].rstrip(')')
+
+    commit_summary = message_parts[1]
+
+    # Check for invalid commit 'type'
+    types = regex_types(args)
+    if commit_type not in types.split(', '):
+        error = ERROR_TYPE.format(types)
+        raise_error(message_title, error, types, args)
+
+    # If 'scope' is provided, check for valid 'scope'
+    if commit_scope and not re.match(r'^[a-z][a-zA-Z0-9_-]*$', commit_scope):
+        raise_error(message_title, ERROR_SCOPE_CAPITALIZATION, types, args)
+
+    # Check for valid length of 'summary'
+    summary_length = len(commit_summary)
+    if summary_length < args.subject_min_length or summary_length > args.subject_max_length:
+        error = ERROR_SUMMARY_LENGTH.format(args.subject_min_length, args.subject_max_length)
+        raise_error(message_title, error, types, args)
+
+    # Check if the 'summary' starts with a lowercase letter
+    if commit_summary[0].islower():
+        raise_error(message_title, ERROR_SUMMARY_CAPITALIZATION, types, args)
+
+    # Check if the 'summary' ends with a period (full stop)
+    if commit_summary[-1] == '.':
+        raise_error(message_title, ERROR_SUMMARY_PERIOD, types, args)
+
+    # Parse commit message additional 'body' (if it exists)
+    if message_body:
+        if re.match(r'\n', message_body[0]):
+            body = message_body[0][1:]  # Remove the first blank line
+
+            # Check if each line of the 'body' is no longer than 100 characters
+            lines = body.split('\n')
+            invalid_lines = [line for line in lines if len(line) > args.body_max_line_length]
+            num_invalid_lines = len(invalid_lines)
+            if num_invalid_lines:
+                error = ERROR_BODY_LENGTH.format(num_invalid_lines)
+
+                for line in invalid_lines:
+                    print(f"  Line: '{line[:30]}...' (length: {len(line)})")
+                raise_error(message_title, error, types, args)
+        else:
```

Comment:
@radimkarnis I agree with the guard idea to avoid nesting, thank you. 

However, `message_body` is optional, so its absence should not cause an error.
The suggestion has been implemented with respect to this fact.

Reply:
You are right, I have missed that, thanks for the explanation!